[ADD] maintain: connect office365 with azure oauth (#3221)
odoo/documentation: content/administration/maintain/azure_oauth.rst

=======================================================
Connect Microsoft Outlook 365 to Odoo using Azure OAuth
=======================================================

Odoo is compatible with Microsoft's Azure OAuth for Microsoft 365. In order to send and receive
secure emails from a custom domain, all that is required is to configure a few settings on the
Azure platform and on the back end of the Odoo database. This configuration works with either a
personal email address or an address created by a custom domain.

> .. tip::
>    For more information, visit `Microsoft's documentation <https://learn.microsoft.com/en-us/azure/
>    active-directory/develop/quickstart-register-app>`_ on setting up OAuth in Azure.

Reviewer: - If the tip is to suggest opening a link, then it's rather a seealso
- In RST, you can go to another line wherever you can write a space. So not in the middle of a link ^^
- Remove the region/language part of your URLs whenever possible (here the `/en-us`). This way, the link opens correctly for everyone, according to their cached settings on Microsoft's website or localization. 
```suggestion
.. seealso::
   `Microsoft Learn: Register an application with the Microsoft identity platform
   <https://learn.microsoft.com/azure/active-directory/develop/quickstart-register-app>`_ 
```

Reviewer: It's just a single comment here, not a full review

Reviewer 2: Hi @everyone.

Just a quick note while doing routine checks.

We can safely break in the middle of a hyperlink URL address as long as it's not in a tag like `:ref:` `:doc:` etc.

From the [RST guidelines](https://www.odoo.com/documentation/15.0/contributing/documentation/rst_guidelines.html#start-a-new-line-before-the-100th-character):
```
You can safely break a line around the separators (-->) of menuselection markups and anywhere in a
hyperlink reference. For the doc, ref and download markups, this is only true for the label part of
the reference.
```
This is confirmed by running a quick `make html` and seeing if the hyperlink breaks (which it shouldn't 😉). I can provide a number of live examples of where we did this on previous docs, if needed.

I do like JCS suggestion here though on turning this into a `:seealso:` which also allows for more context on the link before clicking it. Vague "click here" links aren't actually that helpful 😅

:guilabel:`API permissions` link, located in the left menu bar. Next, click on the
:guilabel:`(+) Add a Permission` button and select :guilabel:`Microsoft Graph` under
:guilabel:`Commonly Used Microsoft APIs`. After, select the :guilabel:`Delegated Permissions`
option.

In the search bar, search for the following :guilabel:`Deregulated permissions` and click
:guilabel:`Add permissions` for each one:

- :guilabel:`SMTP.Send`
- :guilabel:`IMAP.AccessAsUser.All`

.. note::
   The :guilabel:`User.Read` permission will be added by default.

.. image:: azure_oauth/permissions.png
   :align: center
   :alt: API permissions added for the app.

Assign users and groups
=======================

After adding the API permissions, navigate back to the :guilabel:`Overview` of the
:guilabel:`Application` in the top of the left sidebar menu.

Now, add users to this application. Under the :guilabel:`Essentials` overview table, click on the
link labeled :guilabel:`Managed Application in Local Directory`, or the last option on the bottom
right-hand side of the table.

.. image:: azure_oauth/managed-application.png
   :align: center

>    :alt: Link to the created application and to add users/groups.

Author: ```suggestion
   :alt: Add users/groups by clicking the Managed application in local directory link for the created application.
```

application.

Create credentials
------------------

Now that the Microsoft Azure app is set up, credentials need to be created for the Odoo setup.
These include the :guilabel:`Client ID` and :guilabel:`Client Secret`. To start, the
:guilabel:`Client ID` can be copied from the :guilabel:`Overview` page of the app. The
:guilabel:`Client ID` or :guilabel:`Application ID` is located under the :guilabel:`Display Name`
in the :guilabel:`Essentials` overview of the app.

.. image:: azure_oauth/application-id.png
   :align: center
   :alt: Application/Client ID located in the Overview of the app.

Next, the :guilabel:`Client Secret Value` needs to be retrieved. To get this value, click on
:guilabel:`Certificates & Secrets` in the left sidebar menu. Then, a :guilabel:`Client Secret`
needs to be produced. In order to do this, click on the :guilabel:`(+) New Client Secret` button.

A window on the right will populate with a button labeled :guilabel:`Add a client secret`. Under
:guilabel:`Description`, type in `Odoo Fetchmail` or something recognizable, and then set the
:guilabel:`expiration date`.

.. important::

>    A new :guilabel:`Client Secret` will need to be produced and configured if the first one expires.
>    In this event, there could be an interruption of service, so the expiration date should be noted
>    and set to the furthest possible date.

Reviewer 2: Need to break line 107 sooner, as it's 101 characters.
```suggestion
   A new :guilabel:`Client Secret` will need to be produced and configured if the first one
   expires. In this event, there could be an interruption of service, so the expiration date should
   be noted and set to the furthest possible date.
```

<email_communication/default_from>`.

Then, cick on :guilabel:`Connect your Outlook account`.

A new window from Microsoft opens to complete the :guilabel:`authorization process`. Select the
appropriate email address that is being configured in Odoo.

.. image:: azure_oauth/verify-outlook.png
   :align: center
   :alt: Permission page to grant access between newly created app and Odoo.

Then, allow Odoo to access the Microsoft account by clicking on :guilabel:`Yes`. After this, the
page will navigate back to the newly configured :guilabel:`Outgoing Mail Server` in Odoo. The
configuration automatically loads the :guilabel:`token` in Odoo, and a tag stating
:guilabel:`Outlook Token Valid` appears in green.

.. image:: azure_oauth/outlook-token.png
   :align: center
   :alt: Valid Outlook Token indicator.

Finally, click :guilabel:`Test Connection`. A confirmation message should appear. The Odoo database
can now send safe, secure emails through Microsoft Outlook using OAuth authentication.

Multiple user configuration
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Each user should have a separate server set up. The :guilabel:`from-filter` should be set so that

> only the user's email will be sent from that server. In other words, only a user with an email
> address that matches the set :guilabel:`from-filter` will be able to use this server.

Reviewer 3: ```suggestion
only the user's email is sent from that server. In other words, only a user with an email
address that matches the set :guilabel:`from-filter` is able to use this server.
```

Reviewer 2: Resolved on [1171cff](https://github.com/odoo/documentation/pull/3221/commits/1171cffa7cee2fe708f8ccaef6c9fe658916599a) 👍

> Each user should have a separate server setup and the :guilabel:`from-filter` should be set so that
> only the user's email will be sent from that server. In other words, only a user with a matching
> email address will be able to use this server.
> 
> Then, set up a fall-back account email to allow for the sending of :guilabel:`notifications`. This
> email would be configured as a :guilabel:`general transactional server`. The
> :guilabel:`mail.default.from` system parameter would need to be set to the :guilabel:`username` of
> the general transactional server account. For more information see :ref:`Use a default email
> address <email_communication/email_domain/default>`.

Reviewer 3: ```suggestion
Each user should have a separate server set up. The :guilabel:`from-filter` should be set so that
only the user's email will be sent from that server. In other words, only a user with an email 
address that matches the set :guilabel:`from-filter` will be able to use this server.

After setting the :guilabel:`from-filter`, set up a fallback email account to allow for the sending
of :guilabel:`notifications`. The fallback email must be configured as a :guilabel:`general
transactional server`. The :guilabel:`mail.default.from` system parameter must be set to the
:guilabel:`username` of the general transactional server account. For more information see :ref:`Use
a default email address <email_communication/email_domain/default>`.
```

> address <email_communication/email_domain/default>`.

Reviewer 4: ```suggestion
address <email_domain/default>`.
```
